core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -98,11 +99,25 @@ class KafkaApis(val requestChannel: RequestChannel,
       request.apiLocalCompleteTimeMs = SystemTime.milliseconds
   }
 
+  //TODO: this method should only use request.header after all the requests are migrated to use client java request class.
```

Comment:
Minor, but passing in -1 as the apiVersion for JoinGroupRequest and HeartbeatRequest briefly threw me off.

It might be a bit clearer to split up this method into a kafka.api request version validator and a org.apache.kafka.common.requests request version validator. Or maybe make apiVersion an optional parameter.